kiwi/boot/functions.sh

```diff
@@ -5525,18 +5540,25 @@ function waitForStorageDevice {
     # which could be a whole disk or a partition.
     # the function will wait until the size of the
     # storage device could be obtained or the check
-    # counter equals 4
+    # counter equals 30. For USB devices it waits
+    # only until check counter equals 2.
     # ----
     local IFS=$IFS_ORIG
     local device=$1
     local check=0
+    local transport="`getDeviceTransportType $device`"
+    if [[ $transport == *"usb"* ]];then
```

Comment:
you could also consider to code a case statement even though we will match only one case it might be nicer to maintain when more transport types follows

```bash

case "$transport" in
    "usb")
        # handle usb transport type
    ;;
    *)
        # handle default transport
    ;;
    esac
```

I leave that up to you